attiny84_sketch/pidog2/pidog2.ino

```diff
@@ -157,14 +167,25 @@ void doSecondWork() {
 
     rf.set(REG_STATUS,s);
 
-    if (true) {
-        // NB the power pin has negative polarity
-        digitalWrite(PIN_PWR,      !(s & _BV(STAT_PWR_ON)));
+    // Only write to the power status registers if needed.
+    if (pwr_state != bool(s & _BV(STAT_PWR_ON))) {
```

Comment:
Maybe pull this into its own function to simplify this logic:

```c
// feather creates a "soft" start by pwm'ing the power pin from
// completely off to completely on. This reduces inrush current,
// and consequently reduces voltage dip on the 5V rail. Your
// application many or may not benefit from soft-start depending
// on the stiffness of your 5V supply and the load and capacitance
// on the switched side.
void setPiPower(bool on, bool feather) {
    if (on) {
        if (feather) {
            uint32_t dummy = 0;
            for (uint8_t i=0;i<255;i++) {
                PORTA &= B01111111; //ON
                for (uint8_t j=0;j<i;j++) { dummy++; } //Increasing delay
                PORTA |= B10000000; //OFF
                for (uint8_t j=i;j<255;j++) { dummy++; } //Decreasing delay
            }
            PORTA &= B01111111; // Leave it ON
        } else {
            digitalWrite(PIN_PWR,0);
        }
    } else {
        digitalWrite(PIIN_PWR,0);
    }
}
````

Then we can just do

```c
setPiPower(s & _BV_STAT_PWR_ON, s & _BV(STAT_SOFT_START));
```

Reply:
Done. Just need to test changes.

Reply:
Changes test okay.